packages/ai/src/Button.ts

```diff
@@ -152,28 +263,33 @@ class Button extends UI5Element {
 	async _fadeOut(): Promise<void> {
 		const fadeOutDuration = 180;
 
-		const button = this._mainButton;
+		const button = this._splitButton;
 		const newStateObject = this._effectiveStateObject;
 
 		if (!newStateObject) {
 			// eslint-disable-next-line no-console
 			console.warn(`State with name="${this.state}" doesn't exist!`);
 		} else if (button) {
 			const buttonWidth = button.offsetWidth;
-			const hiddenButton = this.shadowRoot?.querySelector(".ui5-ai-button-hidden") as MainButton;
-			button.style.width = `${buttonWidth}px`;
+			const hiddenButton = this.shadowRoot?.querySelector(".ui5-ai-button-hidden") as SplitButton;
+
+			this.buttonToMenu = (!this._currentStateObject?.splitMode && newStateObject?.splitMode) || (!this._currentStateObject?.endIcon && !!newStateObject?.endIcon);
```

Comment:
We could do it like that for readibility and avoid redunant nullish access. 
The `Partial<T>` type makes all props of `<T>` optional.

```ts
		const currentState: Partial<ButtonState> = this._currentStateObject || {};
		this.buttonToMenu = (!currentState.splitMode && newStateObject.splitMode) || (!currentState.endIcon && !!newStateObject.endIcon);
		this.menuToButton = (currentState.splitMode && !newStateObject.splitMode) || (!!currentState.endIcon && !newStateObject.endIcon);
```

Reply:
Done.